### PR TITLE
feat(quickstart): Remove kafka-setup as a hard deployment requirement

### PR DESCRIPTION
**Pull request by pedro93**
This PR does a few improvements to our docker cli behaviour, namely:
 - Makes `kafka-setup` job optional behind the flag: `--kafka-setup` which works similar to standalone consumer containers.
 - Makes docker-compose Quickstart generation logic able to handle duplicate env vars with last-wins heuristic. This was done because we have conflicting configs coming from docker env files & explicit docker-compose environment variables.
 - Generates Neo4j M1-compatible docker-compose files which removes a long-standing limitation in our cli.
 - M1 Quickstart was not being generated by our internal scripts. This makes it very easy for changes in our partial compose files to not get mirrored in the final Quickstart file. This has now been fixed.
 - Makes standalone consumer configs for neo4j deployments consistent.
 - Improves Quickstart comparison logic more flexible with the use of `yq`
 - Bumps Confluent dependencies to 7.2.2

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docker/quickstart/docker-compose.consumers-without-neo4j.quickstart.yml [hunk 1]
```diff
@@ -43,10 +41,6 @@ services:
     - ENTITY_REGISTRY_CONFIG_PATH=/datahub/datahub-mce-consumer/resources/entity-registry.yml
     - DATAHUB_SYSTEM_CLIENT_ID=__datahub_system
     - DATAHUB_SYSTEM_CLIENT_SECRET=JohnSnowKnowsNothing
-    - ENTITY_SERVICE_ENABLE_RETENTION=true
```
Comment (david-leifker) [inline docker/quickstart/docker-compose.consumers-without-neo4j.quickstart.yml:46]: I am concerned about these being removed and the two DATAHUB_* ones above.

File: docker/quickstart/docker-compose.consumers.quickstart.yml [hunk 2]
```diff
@@ -31,32 +31,22 @@ services:
...
-    - NEO4J_PASSWORD=datahub
-    - JAVA_OPTS=-Xms1g -Xmx1g
     - GRAPH_SERVICE_DIFF_MODE_ENABLED=true
-    - GRAPH_SERVICE_IMPL=neo4j
+    - GRAPH_SERVICE_IMPL=elasticsearch
+    - JAVA_OPTS=-Xms1g -Xmx1g
     - ENTITY_REGISTRY_CONFIG_PATH=/datahub/datahub-mce-consumer/resources/entity-registry.yml
     - DATAHUB_SYSTEM_CLIENT_ID=__datahub_system
     - DATAHUB_SYSTEM_CLIENT_SECRET=JohnSnowKnowsNothing
-    - ENTITY_SERVICE_ENABLE_RETENTION=true
```
Comment (david-leifker) [inline docker/quickstart/docker-compose.consumers.quickstart.yml:56]: Same here, must be missing these from some other file and then dropped when regenerated.

**Comment by david-leifker** [comment]
Like the approach, probably you've found an issue with the mce consumer quickstart generation.

File: docker/quickstart/docker-compose.consumers-without-neo4j.quickstart.yml (hunk 1 above)
Comment (pedro93) [reply docker/quickstart/docker-compose.consumers-without-neo4j.quickstart.yml:46]: I can omit the Quickstart generation if needed

Comment (pedro93) [reply docker/quickstart/docker-compose.consumers-without-neo4j.quickstart.yml:46]: The issue here is that these env vars were manually put in these files: https://github.com/datahub-project/datahub/pull/6744/files#r1055954698 when they should have been put in non-quick start docker-compose files and these files generated by the `generate_docker_quickstart.sh` file

File: docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml [hunk 3]
```diff
@@ -16,33 +16,23 @@ services:
     - KAFKA_HEAP_OPTS=-Xms256m -Xmx256m
     - KAFKA_CONFLUENT_SUPPORT_METRICS_ENABLE=false
     hostname: broker
-    image: confluentinc/cp-kafka:7.2.2
+    image: kymeric/cp-kafka:latest
```
Comment (david-leifker) [inline docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml:19]: This is not correct, this repo is like 3 years old at this point and the confluentinc/cp-kafka has support for arm now.

File: docker/quickstart/docker-compose-m1.quickstart.yml [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    - KAFKA_BROKER_ID=1
+    - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
+    - KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+    - KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092
+    - KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1
+    - KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0
+    - KAFKA_HEAP_OPTS=-Xms256m -Xmx256m
+    - KAFKA_CONFLUENT_SUPPORT_METRICS_ENABLE=false
+    hostname: broker
+    image: kymeric/cp-kafka:latest
```
Comment (david-leifker) [inline docker/quickstart/docker-compose-m1.quickstart.yml:19]: We should not be using this image

File: docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml [hunk 5]
```diff
@@ -176,9 +157,9 @@ services:
     - broker
     environment:
     - SCHEMA_REGISTRY_HOST_NAME=schemaregistry
-    - SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS=PLAINTEXT://broker:29092
+    - SCHEMA_REGISTRY_KAFKASTORE_CONNECTION_URL=zookeeper:2181
     hostname: schema-registry
-    image: confluentinc/cp-schema-registry:7.2.0
+    image: eugenetea/schema-registry-arm64:latest
```
Comment (david-leifker) [inline docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml:162]: We do not need this image, the 7.2.x supports arm

File: docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml [hunk 6]
```diff
@@ -187,7 +168,7 @@ services:
     - ZOOKEEPER_CLIENT_PORT=2181
     - ZOOKEEPER_TICK_TIME=2000
     hostname: zookeeper
-    image: confluentinc/cp-zookeeper:7.2.2
+    image: kymeric/cp-zookeeper:latest
```
Comment (david-leifker) [inline docker/quickstart/docker-compose-without-neo4j-m1.quickstart.yml:171]: Same as above

File: docker/quickstart/docker-compose-without-neo4j.quickstart.yml [hunk 7]
```diff
@@ -5,195 +5,172 @@ services:
...
+      - KAFKA_BROKER_ID=1
+      - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
+      - KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      - KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092
+      - KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1
+      - KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0
+      - KAFKA_HEAP_OPTS=-Xms256m -Xmx256m
+      - KAFKA_CONFLUENT_SUPPORT_METRICS_ENABLE=false
     hostname: broker
     image: confluentinc/cp-kafka:5.4.0
```
Comment (david-leifker) [inline docker/quickstart/docker-compose-without-neo4j.quickstart.yml:19]: This is old, can we be consistent on 7.2.x?

File: docker/quickstart/docker-compose.quickstart.yml [hunk 8]
```diff
@@ -5,216 +5,194 @@ services:
...
+      - KAFKA_BROKER_ID=1
+      - KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181
+      - KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      - KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092
+      - KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR=1
+      - KAFKA_GROUP_INITIAL_REBALANCE_DELAY_MS=0
+      - KAFKA_HEAP_OPTS=-Xms256m -Xmx256m
+      - KAFKA_CONFLUENT_SUPPORT_METRICS_ENABLE=false
     hostname: broker
     image: confluentinc/cp-kafka:5.4.0
```
Comment (david-leifker) [inline docker/quickstart/docker-compose.quickstart.yml:19]: Consistent image versions

**Comment by david-leifker** [comment]
I am wondering if we even need a separate m1 flavor anymore? Seems like folks have moved to generate arm images consistently, however perhaps I am overlooking something here.

**Comment by pedro93** [comment]
> I am wondering if we even need a separate m1 flavor anymore? Seems like folks have moved to generate arm images consistently, however perhaps I am overlooking something here.

There are a few images that we need M1 alternatives for still such as Postgres & Neo4j.

**Review by david-leifker — APPROVED** [review]

**Pull request merged**